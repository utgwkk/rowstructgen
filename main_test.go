package main

import (
	_ "embed"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/k0kubun/sqldef/database"
	"github.com/k0kubun/sqldef/parser"
)

//go:embed testdata/schema.sql
var schema string

func prepareDDL(t *testing.T) *parser.DDL {
	mysqlParser := database.NewParser(parser.ParserModeMysql)
	ddls, err := mysqlParser.Parse(schema)
	if err != nil {
		t.Fatal(err)
	}

	ddl, err := extractTableDefinition(ddls, Options{
		Table: "users",
	})
	if err != nil {
		t.Fatal(err)
	}

	return ddl
}

func TestConvertDDLToStructDef(t *testing.T) {
	ddl := prepareDDL(t)
	opts := ConvertOptions{
		PackageName: "dbrow",
		StructName: "User",
	}
	code, err := convertDDLToStructDef(ddl, opts)
	if err != nil {
		t.Fatal(err)
	}
	snaps.MatchSnapshot(t, code)
}
