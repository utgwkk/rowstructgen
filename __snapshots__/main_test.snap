
[TestConvertDDLToStructDef/default - 1]
package dbrow

import "time"

type User struct {
    Id        uint64     `db:"id"`
    Name      string     `db:"name"`
    DeletedAt *time.Time `db:"deleted_at"`
    CreatedAt time.Time  `db:"created_at"`
    UpdatedAt time.Time  `db:"updated_at"`
}

---
